guides/source/layout.html.erb

```diff
@@ -39,59 +51,83 @@
         <li class="more-info"><a href="https://github.com/rails/rails">Contribute on GitHub</a></li>
       </ul>
     </div>
-  </div>
-  <div id="header">
+  </nav>
+  <header id="page_header">
     <div class="wrapper clearfix">
-      <div class="header-logo"><a href="index.html" title="Return to home page">Guides.rubyonrails.org</a></div>
-      <ul class="nav">
-        <li><a class="nav-item" href="index.html">Home</a></li>
-        <li class="guides-index guides-index-large">
-          <a href="index.html" id="guidesMenu" class="guides-index-item nav-item">Guides Index</a>
-          <div id="guides" class="clearfix" style="display: none;">
-            <hr />
-            <dl class="guides-section-container">
-              <% documents_by_section.each do |section| %>
-                <div class="guides-section">
-                  <dt><%= section['name'] %></dt>
+      <nav id="feature_nav"> <%# we use nav as an element to also provide the aria role and improve semantic meaning. %>
+        <div class="header-logo">
+          <a href="index.html" title="Return to home page">Guides</a>
+          <span id="version_switcher">
+            Version:
+            <select class="guides-version">
+              <option value="https://edgeguides.rubyonrails.org/" selected>Edge</option>
+              <option value="https://guides.rubyonrails.org/v7.1">7.1</option>
+              <option value="https://guides.rubyonrails.org/v7.0">7.0</option>
+              <option value="https://guides.rubyonrails.org/v6.1">6.1</option>
+              <option value="https://guides.rubyonrails.org/v6.0">6.0</option>
+              <option value="https://guides.rubyonrails.org/v5.2">5.2</option>
+              <option value="https://guides.rubyonrails.org/v5.1">5.1</option>
+              <option value="https://guides.rubyonrails.org/v5.0">5.0</option>
+              <option value="https://guides.rubyonrails.org/v4.2">4.2</option>
+              <option value="https://guides.rubyonrails.org/v4.1">4.1</option>
+              <option value="https://guides.rubyonrails.org/v4.0">4.0</option>
+              <option value="https://guides.rubyonrails.org/v3.2">3.2</option>
+              <option value="https://guides.rubyonrails.org/v3.1">3.1</option>
+              <option value="https://guides.rubyonrails.org/v3.0">3.0</option>
+              <option value="https://guides.rubyonrails.org/v2.3">2.3</option>
+            </select>
+          </span>
+        </div>
+        <ul class="nav">
+          <li><a class="nav-item" id="home_nav" href="https://rubyonrails.org/">Home</a></li>
+          <li class="guides-index guides-index-large">
+            <a href="index.html" id="guidesMenu" class="guides-index-item nav-item">Guides Index</a>
```

Comment:
On larger screens this seems to be slightly off:

![Screenshot 2024-03-18 at 11 11 28](https://github.com/rails/rails/assets/26328/dcc81793-328c-48f5-b04b-4812ea79faf4)

This is on my external monitor, QHD resolution (2560 x 1440 pixels), both Safari and Chrome.

It also seems to swap to a 2-column list for the guides index, which makes the first column go all the way to the end and overflow as you reduce the height.

<details><summary>2-column overflow sample</summary>
<p>

![Screenshot 2024-03-18 at 11 14 26](https://github.com/rails/rails/assets/26328/3e31208b-7cc6-464d-854a-961d89894a80)

</p>
</details> 

Resizing the window decreasing height eventually swaps it to a 3-column layout, which makes all the content/links show properly. Can we keep the 3-column layout on larger versions?

Reply:
<img width="2289" alt="Screenshot 2024-03-18 at 11 08 15 AM" src="https://github.com/rails/rails/assets/2223/fd035fff-91dd-4a49-b303-d0a8a669d854">

Forcing the height based on em - 3 column maintains on 2560x1440, even at full screen

Reply:
Thanks, the 3-column layout seems to be working now.

I'm still seeing that bit of weirdness with the top of the guides index overlay though, on the larger screen. It's actually moving as you resize, causing it to move down enough that the menu elements under it show up:

https://github.com/rails/rails/assets/26328/fd74e232-7158-4f7c-8367-ee9740459187

Reply:
Ah that's what `vw` will do for ya. Changed to explicit pixel values in c99d716b and it holds its place to the upper right.